src/pages/HawkersPage/HawkersPage.tsx

```diff
@@ -43,6 +45,18 @@ const HawkerLocationPage: React.FunctionComponent = () => {
     }
   }
 
+  //Pagination
+  const hawkersPerPage = 5;
+  const pagesVisited = pageNumber * hawkersPerPage;
+  const pageNum = Math.ceil(hawkers.length / hawkersPerPage);
+
+  const hawkerPagination = hawkers.slice(pagesVisited, pagesVisited + hawkersPerPage);
+  const filteredPagination = filteredHawkers.slice(pagesVisited, pagesVisited + hawkersPerPage);
+
+  const changePage = (event: any) => {
+    setPageNumber(event.target.id);
```

Comment:
If you do a `console.log` you will realise `event.target.id` gives you nothing. The correct value is `event.target.getAttribute('value') - 1` (To get this, you can first `console.log(event.target)` to see what is inside). Note that page number starts from 1. Alternatively, you can take a look at `SearchPage.tsx` because another `Pagination` is used there.

Reply:
@niqiukun thank you!